### PR TITLE
Sort API Routes code

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:1]: With this reordering we split features apart in the file. What if we group API calls by feature and have "basic blocks" that are used by multiple features in its own group above everyone else?

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:1]: Maybe not all of them, just calendar?

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt [hunk 1]
```diff
@@ -24,99 +24,133 @@ import com.infomaniak.mail.utils.Utils
...
-    fun draft(mailboxUuid: String) = "$MAIL_API/api/mail/$mailboxUuid/draft"
+    fun flushFolder(mailboxUuid: String, folderId: String): String {
+        return "${folder(mailboxUuid, folderId)}/flush"
+    }
 
-    fun draft(mailboxUuid: String, remoteDraftUuid: String) = "${draft(mailboxUuid)}/$remoteDraftUuid"
+    fun search(mailboxUuid: String, folderId: String, filters: String): String {
+        return "${folder(mailboxUuid, folderId)}/message?thread=on&offset=0&$filters"
+    }
+    //region Folder
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:127]: Typo.

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt [hunk 2]
```diff
@@ -24,99 +24,133 @@ import com.infomaniak.mail.utils.Utils
...
-    private fun messages(mailboxUuid: String) = "$MAIL_API/api/mail/$mailboxUuid/message"
+    //region Mailbox
+    private fun mailbox(): String {
+        return "$MAIL_API/api/mailbox"
+    }
 
-    private fun message(mailboxUuid: String, folderId: String, shortUid: Int): String {
-        return "${folder(mailboxUuid, folderId)}/message/$shortUid"
+    fun mailboxes(): String {
+        return "${mailbox()}?with=aliases,unseen"
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:95]: It's supposed to disappear, yes.
But… not in this PR 🤣 
I've put it back.

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:1]: Problem arises in later PR, fixed there

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt:1]: Fixed in https://github.com/Infomaniak/android-kMail/pull/1665

File: app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt [hunk 3]
```diff
@@ -127,31 +161,86 @@ object ApiRoutes {
         return "${endpoint}${messages}${offset}${direction}"
```
[the remark this answers is not in the document]
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt]: It seems you didn't remove the curly braces

[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/api/ApiRoutes.kt]: Oh yes sorry, I added them on purpose, for readability.

**Review by FabianDevel — APPROVED** [review]

**Review by LunarX — APPROVED** [review]

**Pull request merged**